src/vpx/expanded.rs

```diff
@@ -50,6 +50,141 @@ pub fn extract(vpx_file_path: &Path, expanded_path: &Path) -> std::io::Result<()
+        let ext = img.ext();
+
+        jpeg_path.push(format!("{}.{}", img.name, ext));
+
+        files.push(jpeg_path.to_string_lossy().to_string());
+    }
+    if images_size == 0 {
+        files.push(
+            images_path
+                .join(std::path::MAIN_SEPARATOR_STR)
+                .to_string_lossy()
+                .to_string(),
+        );
+    }
+
+    let sounds_size = gamedata.sounds_size;
+    let sounds_path = root_dir_path.join("sounds");
+    for index in 0..sounds_size {
+        let path = format!("GameStg/Sound{}", index);
+        let mut input = Vec::new();
+        comp.open_stream(&path)
+            .unwrap()
+            .read_to_end(&mut input)
+            .unwrap();
+        let mut reader = BiffReader::new(&input);
+        let sound = sound::read(&version, &mut reader);
+
+        let ext = sound.ext();
+        let mut sound_path = sounds_path.clone();
+        sound_path.push(format!("{}.{}", sound.name, ext));
+
+        files.push(sound_path.to_string_lossy().to_string());
+    }
+    if sounds_size == 0 {
+        files.push(
+            sounds_path
+                .join(std::path::MAIN_SEPARATOR_STR)
+                .to_string_lossy()
+                .to_string(),
+        );
+    }
+
+    let fonts_size = gamedata.fonts_size;
+    let fonts_path = root_dir_path.join("fonts");
+    for index in 0..fonts_size {
+        let path = format!("GameStg/Font{}", index);
+        let mut input = Vec::new();
+        comp.open_stream(&path)
+            .unwrap()
+            .read_to_end(&mut input)
+            .unwrap();
+        let font = font::read(&input);
+
+        let ext = font.ext();
+        let mut font_path = fonts_path.clone();
+        font_path.push(format!("Font{}.{}.{}", index, font.name, ext));
+
+        files.push(font_path.join("/").to_string_lossy().to_string());
+    }
+    if fonts_size == 0 {
+        files.push(fonts_path.to_string_lossy().to_string());
+    }
+
+    let entries = retrieve_entries_from_compound_file(&mut comp);
+    entries.iter().for_each(|path| {
+        let mut stream = comp.open_stream(path).unwrap();
+        // write the steam directly to a file
+        let file_path = root_dir_path.join(&path[1..]);
+        // println!("Writing to {}", file_path.display());
+        files.push(file_path.to_string_lossy().to_string());
+    });
+
+    files.sort();
+
+    // These files are made by:
+
+    // -extract_script
+    files.push(
+        root_dir_path
+            .join("script.vbs")
+            .to_string_lossy()
+            .to_string(),
+    );
+    // -extract_collections
+    files.push(
+        root_dir_path
+            .join("collections.json")
+            .to_string_lossy()
+            .to_string(),
+    );
+    // -extract_info
+    files.push(
+        root_dir_path
+            .join("TableInfo.json")
+            .to_string_lossy()
+            .to_string(),
+    );
+    // TODO -extract_gameitems
+
+    for file_path in files {
```

Comment:
can you keep console io in main?

So returning the files here and println in main, that way we can test this function